packages/utils/src/lib/raf.ts

```diff
@@ -6,26 +6,101 @@ const isTest = () =>
 
 const rafQueue: Array<() => void> = []
 
+type RenderingMode = 'raf' | 'sixtyFps'
+type UpdateMode = 'onTick' | 'onPointerMove'
+type ReactUpdateMode = 'original' | 'throttled'
+
+// How do we schedule the updates? Either every 16ms or on raf
+/** @internal */
+export const renderingMode: RenderingMode = 'sixtyFps'
+// export const renderingMode: RenderingMode = 'raf'
+// How does we process the updates? Either onTick or onPointerMove
+/** @internal */
+export const updateMode: UpdateMode = 'onTick'
+// export const updateMode: UpdateMode = 'onPointerMove'
+// How do we notifiy React about updates? Either original (as soon as the change occurs) or throttled
+/** @internal */
+export const reactUpdateMode: ReactUpdateMode = 'throttled'
+// export const reactUpdateMode: ReactUpdateMode = 'original'
+
+let timesBetweenFrames: number[] = []
+let lastFrameTime = 0
+
 const tick = () => {
+	const now = Date.now()
+	const timeSinceLastTick = now - lastFrameTime
+	if (timeSinceLastTick < 1000) {
+		timesBetweenFrames.push(timeSinceLastTick)
+	}
+	if (timesBetweenFrames.length > 100) {
+		// eslint-disable-next-line no-console
+		console.log('resetting fps')
+		timesBetweenFrames = []
+	}
+	const average = timesBetweenFrames.reduce((a, b) => a + b, 0) / timesBetweenFrames.length
+	const fps = `${Math.round(1000 / average)}fps`
+	// eslint-disable-next-line no-console
+	console.log('time since last tick', timeSinceLastTick, fps)
+	lastFrameTime = now
+
 	const queue = rafQueue.splice(0, rafQueue.length)
 	for (const fn of queue) {
 		fn()
 	}
 }
 
+const fps = 60
```

Comment:
i see down below that the you account for the frame rates falling below the target frame rate.

does starting the target at 60 hurt any machines that can't get that fast? if a machine has more a frame rate like 30, they basically go through the fallback path of lines 82-87, is that right? any downside?

Reply:
I think 60 is a good starting point. This affects all updates, even stuff like dragging the mouse cursor around, which probably can be handled even by slow devices. So I'd say we want that as default.

For those more expensive operations:
- Like mentioned in the PR description the browser starts throttling the calls to RAF callbacks. This happens at around 10-20 fps. So when that happens our code also gets throttled.
- We can also look into the mentioned variable frame rate, which should adjust how much we tax the browser.

That said it's a bit hard to test all of this. Like how does the variable frame rate interact with browser throttling? My guess is that if we adjust the refresh rate we should be taxing the browser less and thus hit the throttling later or maybe not hit it at all in some cases. We might also land in the sweet spot of what the device is able to do - this is probably also what happens due to browser throttling, but unsure 🤷‍♂️ 😂